### ISSUE TITLE
Segmentation ideas

### ISSUE THREAD

**Issue by calebrob6** [torchgeo/torchgeo-bench#8]
From @nilsleh 

A little summary of ideas for segmentation:

- https://arxiv.org/pdf/2304.07193 The DinoV2 paper in 7.4 describes that they have two setups for probing:
  - train linear layer on patch tokesn to produce segmentation class logits and then upsample to target resolution
  - +ms version: concatenate the patch tokens of last 4 layers , also use linear linear but also multi-scale test time agumentations 
  - they are able to get competitive or close performance to fully finetuning MAE with Upernet decoder
  - this could be an approach for the ViT based models and could also try something similar for the Conv based models (already implemented in torchgeo-bench at basic level)
  - it is also used in this recent benchmark foundation model paper https://arxiv.org/abs/2401.11311

- Another Avenue that I have found in foundation model literature is few-shot segmentation via prototype learning, for which this paper is apparantly the OG: https://openaccess.thecvf.com/content_ICCV_2019/html/Wang_PANet_Few-Shot_Image_Semantic_Segmentation_With_Prototype_Alignment_ICCV_2019_paper.html
  - here you extract features with the frozen backbone, compute class prototypes with masked average pooling based on a support set of some labeled examples
  - then for each image you want to make predictions for, compute cosine simlarity of extracted feature to the class prototypes, which then can give you a pixel-wise argmax which is the segmentation prediction
  - I don't have experience with using prototypes, but it sounds interesting as a an alternative "training-free" approach to the small linear probe
  
Maybe one of you has experience with this? Or otherwise opinions? In both cases, you could also run the resizing experiments etc.

If we also want to think along the dimension of how to use RGB Imagenet models for Sentinel 2 13 channel segmentation, we would probably have to resort to a simliar adopter scheme. (edited) 

https://github.com/billpsomas/efficient-probing hot new probing method for ViTs
